### PR TITLE
Update hosts.example to model minimum viable production cluster

### PR DESCRIPTION
**Pull request by sdodson**
Update hosts.example to model minimum viable production cluster.
This includes three masters, two infra nodes, two compute nodes, and
an haproxy load balancer to load balance traffic to the API servers. For
a truly production environment you should use an external load
balancing solution that itself is highly available.

Remaining challenges:
 - [lb] defined load balancer is a single point of failure but it's the only way that we can provision a complete multi-master environment
 - nfs based storage for registry is not supported, but it's the only way that we can provision an HA registry without relying on external object storage.

### CODE REVIEW

**Comment by sdodson** [comment]
CC @jctanner

**Comment by jctanner** [comment]
Are the masters and the nfs node supposed to have different FQDNs?

```
[masters]
ose3-master[1:3]-ansible.test.example.com

[nfs]
ose3-master1.example.com
```

File: inventory/hosts.example [hunk 1]
```diff
@@ -37,6 +53,23 @@ openshift_deployment_type=origin
 # release.
 openshift_release=v3.7
 
+# default subdomain to use for exposed routes, you should have wildcard dns
+# for *.apps.test.example.com that points at your infra nodes which will run
+# your router
+openshift_master_default_subdomain=apps.test.example.com
+
+#Set cluster_hostname to point at your load balancer
+openshift_master_cluster_hostname=ose-lb.test.example.
```
Comment (jctanner) [inline inventory/hosts.example]: this should be "ose-lb.test.example.com", right? When the cluster comes up it's trying to check status on "ose-lb.test.example." and that continuously fails.

Comment (sdodson) [reply inventory/hosts.example]: yep, updated.

**Pull request merged**